### ISSUE TITLE
GlobalMessagePipe.SetProvider does not work with IL2CPP

### ISSUE THREAD

**Issue by YuminekoGame** [Cysharp/MessagePipe#29]
Calling `GlobalMessagePipe.SetProvider (provider)` during IL2CPP Build results in an `InvalidOperationException`.

- Unity 2021.1.16f1
- Windows 10 64bit

### Sample Code

```csharp
using MessagePipe;
using UnityEngine;

public class Installer
{
    [RuntimeInitializeOnLoadMethod]
    private static void Initialize()
    {
        var builder = new BuiltinContainerBuilder();
        builder.AddMessagePipe();
        builder.AddMessageBroker<int>();
        
        var provider = builder.BuildServiceProvider();
        GlobalMessagePipe.SetProvider(provider);
    }
}
```

```
InvalidOperationException: Sequence contains no elements
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00000] in <00000000000000000000000000000000>:0 
  at MessagePipe.ServiceProviderType..ctor (System.Type type) [0x00000] in <00000000000000000000000000000000>:0 
  at MessagePipe.BuiltinContainerBuilderServiceProvider+<>c__DisplayClass2_1.<.ctor>b__1 () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Func`1[TResult].Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Lazy`1[T].CreateValue () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Lazy`1[T].LazyInitValue () [0x00000] in <00000000000000000000000000000000>:0 
  at MessagePipe.BuiltinContainerBuilderServiceProvider.GetService (System.Type serviceType, System.Int32 depth) [0x00000] in <00000000000000000000000000000000>:0 
  at MessagePipe.DependencyInjectionShims.GetRequiredService[T] (System.IServiceProvider provider) [0x00000] in <00000000000000000000000000000000>:0 
  at MessagePipe.GlobalMessagePipe.SetProvider (System.IServiceProvider provider) [0x00000] in <00000000000000000000000000000000>:0 
```

**Comment by neuecc** [comment]
What is your stripping level?

**Comment by neuecc** [comment]
I've tested.

Low - OK
Medium - NG
High - NG

It should be fix, thanks!

**Comment by YuminekoGame** [comment]
Probably Low...
![2021-05-07_13h34_25](https://user-images.githubusercontent.com/15545226/117398253-f518ba80-af38-11eb-833d-586016be0a07.png)

**Comment by neuecc** [comment]
Oh, it's weird.......
Anyway, I have released `1.3.1`.
This is all working in my test environment, even with IL2CPP at Stripping Level High.

**Comment by YuminekoGame** [comment]
I confirmed the correct operation on 1.3.1.
Thank you for your prompt correction!
